diff -Nur SDL_mixer-1.2.6-org/timidity/config.h SDL_mixer-1.2.6/timidity/config.h
--- SDL_mixer-1.2.6-org/timidity/config.h	2004-12-16 14:20:14.000000000 +0900
+++ SDL_mixer-1.2.6/timidity/config.h	2007-08-19 00:12:16.000000000 +0900
@@ -51,7 +51,7 @@
 
 /* Default sampling rate, default polyphony, and maximum polyphony.
    All but the last can be overridden from the command line. */
-#define DEFAULT_RATE	32000
+#define DEFAULT_RATE	22050
 /* #define DEFAULT_VOICES	32 */
 /* #define MAX_VOICES	48 */
 #define DEFAULT_VOICES	256
@@ -112,12 +112,12 @@
 /* On some machines (especially PCs without math coprocessors),
    looking up sine values in a table will be significantly faster than
    computing them on the fly. Uncomment this to use lookups. */
-/* #define LOOKUP_SINE */
+#define LOOKUP_SINE
 
 /* Shawn McHorse's resampling optimizations. These may not in fact be
    faster on your particular machine and compiler. You'll have to run
    a benchmark to find out. */
-#define PRECALC_LOOPS
+/* #define PRECALC_LOOPS */
 
 /* If calling ldexp() is faster than a floating point multiplication
    on your machine/compiler/libm, uncomment this. It doesn't make much
@@ -227,7 +227,7 @@
 #ifdef __WIN32__
 #define DEFAULT_PATH	"\\TIMIDITY"
 #else
-#define DEFAULT_PATH	"/usr/local/lib/timidity"
+#define DEFAULT_PATH	"/etc/"
 #endif
 
 /* These affect general volume */
